packages/eslint-plugin-wpcalypso/lib/rules/test/i18n-unlocalized-url.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file Disallow using unlocalized URL strings
+ * @author Automattic
+ * @copyright 2023 Automattic. All rights reserved.
+ * See LICENSE.md file in root directory for full license.
+ */
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const RuleTester = require( 'eslint' ).RuleTester;
+const rule = require( '../i18n-unlocalized-url' );
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const localizableUrls = [
+	'https://apps.wordpress.com/',
+	'https://automattic.com/cookies/',
+	'https://automattic.com/privacy/',
+	'https://wordpress.com/support/test-article/',
+	'https://wordpress.com/forums/',
+	'https://wordpress.com/tos/',
+];
+
+const nonLocalizableUrls = [ 'https://example.com', 'https://wordpress.com/forums/some-thread' ];
+
+new RuleTester( {
+	parserOptions: {
+		sourceType: 'module',
+		ecmaFeatures: { jsx: true },
+		ecmaVersion: 6,
+	},
+} ).run( 'i18n-unlocalized-url', rule, {
+	valid: [].concat(
+		localizableUrls.reduce( ( cases, url ) => {
+			return cases.concat( [
+				`const url = localizeUrl( '${ url }' )`,
+				`const url = '${ url }'; localizeUrl( url );`,
+				`const link = <a href={ localizeUrl( '${ url }' ) }></a>;`,
+				`const url = '${ url }'; const link = <a href={ localizeUrl( url ) }></a>;`,
+				`const url = localizeUrl( \`${ url }?param=test\` );`,
```

Comment:
Do we support this ternary syntax? Should we add a test case for it?

https://github.com/Automattic/wp-calypso/blob/3c76a54cb4d25de3d3d7ee4ea5a925d54e168bf0/client/blocks/product-purchase-features-list/sell-online-paypal.jsx#L7-L11

Reply:
Good catch! I've added handling for ternary and logical operators in https://github.com/Automattic/wp-calypso/pull/78480/commits/2805a428bed4a53322da2220c445e13b00efb1cd.